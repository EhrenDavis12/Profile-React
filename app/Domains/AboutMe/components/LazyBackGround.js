import PropTypes from 'prop-types';
import React from 'react';
import LazyHero from 'react-lazy-hero';

import BkgImgCalm from 'images/background-calm-clouds.jpg';

function LazyBackGround(props) {
  return (
    <LazyHero
      imageSrc={BkgImgCalm}
      opacity={0}
      parallaxOffset={2500}
      minHeight="290vh" // adjust this number to get more room or to center content
    >
      <div className="container">{props.children}</div>
    </LazyHero>
  );
}

LazyBackGround.propTypes = {
  children: PropTypes.node.isRequired,
};

export default LazyBackGround;
