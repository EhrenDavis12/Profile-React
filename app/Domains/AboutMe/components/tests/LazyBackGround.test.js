import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import LazyBackGround from '../LazyBackGround';

describe('<LazyBackGround />', () => {
  it('should render the page', () => {
    const renderedComponent = shallow(
      <LazyBackGround>
        <div>Hello Wold</div>
      </LazyBackGround>,
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
