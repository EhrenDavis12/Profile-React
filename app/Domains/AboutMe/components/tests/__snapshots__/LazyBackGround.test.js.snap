// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LazyBackGround /> should render the page 1`] = `
<LazyHero
  color="#fff"
  imageSrc="IMAGE_MOCK"
  isCentered={true}
  isFixed={false}
  minHeight="290vh"
  opacity={0}
  parallaxOffset={2500}
  transitionDuration={600}
  transitionTimingFunction="ease-in-out"
>
  <div
    className="container"
  >
    <div>
      Hello Wold
    </div>
  </div>
</LazyHero>
`;
